File: solutions/problem_1129_shortest_path_with_alternating_colors/solution.cpp
/*
 * Problem 1129: Shortest Path with Alternating Colors
 * =================================================
 * Difficulty: Medium
 * Tags: Breadth-First Search, Graph Theory
 * Pattern: BFS Graph Traversal
 *
 * Time Complexity:  O(V + E)
 * Space Complexity: O(V)
 */

#include <queue>
#include <string>
#include <unordered_set>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> shortestAlternatingPaths(int n, vector<vector<int>>& redEdges, vector<vector<int>>& blueEdges) {
        // BFS on graph - O(V+E) time
        if (n.empty()) return {};
        queue<int> q;
        unordered_set<int> visited;
        q.push(0);
        visited.insert(0);
        int dist = 0;
        while (!q.empty()) {
            int sz = q.size();
            for (int i = 0; i < sz; i++) {
                int node = q.front(); q.pop();
                // Process node
            }
            dist++;
        }
        return dist;
    }
};

File: solutions/problem_1129_shortest_path_with_alternating_colors/solution.py
"""
Problem 1129: Shortest Path with Alternating Colors
=================================================
Difficulty: Medium
Tags: Breadth-First Search, Graph Theory
Pattern: BFS Graph Traversal

Time Complexity:  O(V + E)
Space Complexity: O(V)
"""

class Solution:
    def shortestAlternatingPaths(self, n: int, redEdges: List[List[int]], blueEdges: List[List[int]]) -> List[int]:
        # BFS on graph - O(V+E) time
        from collections import deque
        if not n:
            return []
        visited = set()
        queue = deque([0])
        visited.add(0)
        dist = 0
        while queue:
            for _ in range(len(queue)):
                node = queue.popleft()
                # Process node
            dist += 1
        return dist
